packages/vue-language-core/src/generators/script.ts

```diff
@@ -377,8 +377,14 @@ export function generate(
 			);
 			//#endregion
 
+			codes.push(
+				`type OmitKeepDiscriminatedUnion<T, K extends keyof any> = T extends any`,
+  			`? Pick<T, Exclude<keyof T, K>>
+  			: never`,
+				`;\n`,
+			);
```

Comment:
I believe this could be defined in https://github.com/vuejs/language-tools/blob/master/packages/vue-language-core/src/utils/globalTypes.ts since it doesn't depend on any dynamic code.

(I'm not super familiar with this project so you might wait for confirmation on that)

Reply:
you're right, thanks. Done